torch/distributed/_fsdp/fully_sharded_data_parallel.py

```diff
@@ -306,6 +296,41 @@ def __getitem__(self, key: int) -> Any:
         """Forward indexing calls in case the module is a nn.Sequential."""
         return self.module.__getitem__(key)  # type: ignore[operator]
 
+    def __getstate__(self) -> Dict[str, Any]:
```

Comment:
do we need to wrap both `__getstate__` and `__setstate__` under `no_grad` context? Or is it already so?